src/ncp-spinel/SpinelNCPTaskForm.cpp

```diff
@@ -41,17 +41,23 @@ nl::wpantund::SpinelNCPTaskForm::SpinelNCPTaskForm(
 ):	SpinelNCPTask(instance, cb), mOptions(options), mLastState(instance->get_ncp_state())
 {
 	if (!mOptions.count(kWPANTUNDProperty_NetworkPANID)) {
-		uint16_t panid;
+		uint16_t panid = instance->mPanid;
 
-		sec_random_fill(reinterpret_cast<uint8_t*>(&panid), sizeof(panid));
+		if (panid == 0xffff)
```

Comment:
I think we need to initialize the `mPanId` to `0xffff` from constructor to make the un-specified panid in `form` generate the random value.